### DISCUSSION TITLE
Kuvvetini Hesaplama (1)

### DISCUSSION THREAD

**Discussion by EfeMertIrez** [sonsuzus/sonsuzus#340] [Soru Cevap]
### Soru
Verilen iki sayı A ve B. A üssü B'nin değerini hesaplayınız. Çıktı çok büyük olabilir, bu nedenle sonucun modulosunu yazdırın.

### Girdi
Girdi, boşlukla ayrılmış iki tam sayı A ve B içerir.

### Çıktı
A üssü B'nin değerinin modulosunu yazdırın.

Kısıtlamalar:
Test Dosyası 1 ila 5:
1<=A<=10
1<=B<=10

Test Dosyası 6 ila 10:
1<=A<=106
1<=B<=1010

Soru Linki : https://www.hackerearth.com/practice/math/number-theory/basic-number-theory-1/practice-problems/algorithm/calculate-the-power/

**Comment by sonsuzus** [comment]
Modulo ya örnekler verir misin...

**Comment by Talla04** [comment]
```python
def us_alma_mod(A, B, mod):
    if B == 0:
        return 1
    if B % 2 == 0:
        yari_us = us_alma_mod(A, B // 2, mod)
        return (yari_us * yari_us) % mod
    else:
        yari_us = us_alma_mod(A, (B - 1) // 2, mod)
        return (A * yari_us * yari_us) % mod

A, B = map(int, input().split())
mod = 10**9 + 7  # Çıktının modu

sonuc = us_alma_mod(A, B, mod)
print(sonuc)

```
